addons/product/models/product_template.py

```diff
@@ -149,6 +149,18 @@ def _get_default_volume_uom(self):
 
     item_ids = fields.One2many('product.pricelist.item', 'product_tmpl_id', 'Pricelist Items')
 
+    has_configurable_attributes = fields.Boolean("Is a configurable product", compute='_compute_has_configurable_attributes', store=True)
+
+    @api.multi
+    @api.depends('attribute_line_ids')
+    def _compute_has_configurable_attributes(self):
+        """ A product is considered configurable if:
+        - It has dynamic attributes
+        - It has any attribute line with at least 2 attribute values configured"""
+        for product in self:
+            product.has_configurable_attributes = any(attribute.create_variant == 'dynamic' for attribute in product.mapped('attribute_line_ids.attribute_id')) \
```

Comment:
You can use existing `has_dynamic_attributes`.

You might also want to check `_has_is_custom_values`, which also requires user input (not sure what is your goal here).

Reply:
The custom values are defined in sale, so they are not available in the product module scope.
The function is overridden in sale_product_configurator to consider custom values.

Reply:
Ok for custom, makes sense.